platforms/esp32c3-buddy/src/heap.rs

```diff
@@ -38,6 +38,9 @@ impl UnderlyingAllocator for UnderlyingEspHeap {
     ///
     /// May or may not require a call to [UnderlyingAllocator::init()] before the allocator
     /// is actually ready for use.
+    //
+    // TODO: Clippy has a point here as this is misuse prone, but silence for now.
+    #[allow(clippy::declare_interior_mutable_const)]
```

Comment:
it's not really misuse-prone when the constant is used as an initializer; the docs for this lint explicitly note that sometimes a constant is used as an initializer and that in that case, the "misuse-prone" behavior is actually correct: https://rust-lang.github.io/rust-clippy/master/index.html#/declare_interior_mutable_const